### PR TITLE
ADR for release captain

### PR DESCRIPTION
**Pull request by jhmarina**
Proposal of tasks that need to be undertaken by a release captain

Signed-off-by: Jesús Herman-Marina <jherman@suse.com>

### CODE REVIEW

File: docs/decisions/0004-release-captain.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+This team member will be the release captain.
+
+This document defines and agrees on the responsibilities of said release captain.
+
+## Considered Options
+
+### Steps to follow on a release
+
+- Build UI container
+- Build Tools container
```
Comment (giubacc) [inline docs/decisions/0004-release-captain.md]:
```suggestion
- Tag s3gw-tools repository main branch with version: vX.Y.Z
- Build Gateway container
- Push Gateway container's image on ghcr.io/aquarist-labs/s3gw:latest
- Push Gateway container's image on ghcr.io/aquarist-labs/s3gw:vX.Y.Z
```

File: docs/decisions/0004-release-captain.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+release, one team member will be responsible for all release-related tasks. 
+This team member will be the release captain.
+
+This document defines and agrees on the responsibilities of said release captain.
+
+## Considered Options
+
+### Steps to follow on a release
+
+- Build UI container
```
Comment (giubacc) [inline docs/decisions/0004-release-captain.md:20]:
```suggestion
- Tag s3gw-ui repository main branch with version: vX.Y.Z
- Build UI container
- Push UI container's image on ghcr.io/aquarist-labs/s3gw-ui:latest
- Push UI container's image on ghcr.io/aquarist-labs/s3gw-ui:vX.Y.Z
```

File: docs/decisions/0004-release-captain.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+- Push UI container's image on ghcr.io/aquarist-labs/s3gw-ui:latest
+- Push UI container's image on  ghcr.io/aquarist-labs/s3gw-ui:vX.Y.Z
+- Build Gateway container
+- Push Gateway container's image on ghcr.io/aquarist-labs/s3gw:latest
+- Push Gateway container's image on ghcr.io/aquarist-labs/s3gw:vX.Y.Z
+- Update changelog documents
```
Comment (giubacc) [inline docs/decisions/0004-release-captain.md]: this should logically precede build steps

**Review by giubacc — CHANGES_REQUESTED** [review]

File: docs/decisions/0004-release-captain.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+release, one team member will be responsible for all release-related tasks.
+This team member will be the release captain.
+
+This document defines and agrees on the responsibilities of said release captain.
+
+## Considered Options
+
+### Steps to follow on a release
+
+- Tag s3gw-ui repository main branch with version: vX.Y.Z
```
Comment (giubacc) [inline docs/decisions/0004-release-captain.md]: there should be 5 tag indications in the document, 4 to do before the final release note step:

- ceph
- s3gw-tools
- s3gw-ui
- s3gw-charts

1 final tag and contextual release notes:

- s3gw

**Review by giubacc — CHANGES_REQUESTED** [review]

File: docs/decisions/0004-release-captain.md (hunk 4 above)
Comment (jhmarina) [reply docs/decisions/0004-release-captain.md]: ~~I'm not entirely sure we're tagging those repos anymore. I believe we now only tag the `s3gw` repo~~

Comment (jhmarina) [reply docs/decisions/0004-release-captain.md]: ~~In fact the text should probably read `Tag the s3gw repository main branch with version: vX.Y.Z`~~

**Review by giubacc — APPROVED** [review]

**Comment by giubacc** [comment]
@jhmarina  approved, but please, sign your commit.

**Comment by jhmarina** [comment]
> @jhmarina approved, but please, sign your commit.

it looks like they are signed off 

Screenshot 2022-09-01 at 11 59 45

Screenshot 2022-09-01 at 12 00 50

This happened recently with another PR. Not sure why 🤷‍♂️

**Pull request merged**